modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1222,23 +1230,41 @@ private void _validateActive(
 	}
 
 	private void _validateEnableCategorization(
-			boolean enableCategorization, boolean system)
+			boolean enableCategorization, String storageType, boolean system)
 		throws PortalException {
 
 		if (enableCategorization && system) {
 			throw new ObjectDefinitionEnableCategorizationException(
 				"Enable categorization is not allowed for system object " +
 					"definition");
 		}
+
+		if (enableCategorization &&
+			!StringUtil.equals(
+				storageType, ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT)) {
+
+			throw new ObjectDefinitionEnableCategorizationException(
+				"Enable categorization is allowed only for storageType " +
```

Comment:
I found this exception in Objects module:
`throw new ObjectLayoutBoxCategorizationTypeException(
	"Categorization layout box only can be used in object definitions with default storage type");`

When we are creating a message, it does not need to be together the words like "objectDefinition" or "storageType".
Maybe we could rewrite to something like this because it would be more informative: "Enable categorization is allowed only in object definition with default storage type"